internal/datasourcev2/polling_http_request.go

```diff
@@ -174,16 +175,30 @@ func (r *pollingRequester) Request() (*PollingResponse, error) {
 				updates = append(updates, fdv2proto.DeleteObject{Kind: kind, Key: key, Version: version})
 			}
 		case fdv2proto.EventPayloadTransferred:
-			//nolint:godox
-			// TODO: deserialize the state and create a fdv2proto.Selector.
+			r := jreader.NewReader(event.Data)
```

Comment:
Wasn't sure if you preferred decoding like this or if it would be better to just decode straight to a simple struct.